### PR TITLE
Use new git-sdk-arm64 for aarch64

### PR DESCRIPTION
**Pull request by dennisameling**
The brand-new https://github.com/git-for-windows/git-sdk-arm64 is now available. Let's use it in our GitHub Action too, instead of having to reinstall those Pacman dependencies every time.

Test workflow in progress: https://github.com/dennisameling/git/actions/runs/3756640711

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -81,15 +22,19 @@ async function run(): Promise<void> {
       return
     }
 
-    const architectureToDownload =
-      architecture === 'aarch64' ? 'x86_64' : architecture
+    if (architecture === 'aarch64' && flavor !== 'full') {
+      throw new Error(
+        'On aarch64, only the "full" flavor is supported at this time.'
+      )
+    }
```
Comment (dennisameling) [inline main.ts:29]: In order to support other flavors, `please.sh create-sdk-artifact` will need to be updated first to move away from `bitness` first. [Pending final decision](https://github.com/git-for-windows/build-extra/pull/452#discussion_r1053792396) on architecture names that GfW will use.

**Comment by dennisameling** [comment]
Had to tweak some other things (like `osslsigncode` for arm64) but I just had a [successful pipeline](https://github.com/dennisameling/git/actions/runs/3760086097) with this Action and the new `git-sdk-arm64` SDK 🎉 🔥

Some observations:
- The `setup-git-for-windows-sdk` on `aarch64` for the `full` flavor takes [only 1m34s](https://github.com/dennisameling/git/actions/runs/3760086097/jobs/6390868987) to set up (no caching involved!). This is noticeably faster than the [3m24s](https://github.com/dennisameling/git/actions/runs/3726455609/jobs/6319931827) for the "old" situation where it cloned `git-sdk-64` (cached) first and then installed the Pacman packages on every run
- With ~23m for the `pkg` job, `aarch64` is the fastest of them all (~25m for `i686` and ~33m for `x86_64`). Some more details:
  - Building `mingw-w64-clang-aarch64-git` (the main package) takes ~6 mins. All the other minutes ([19m in total](https://github.com/dennisameling/git/actions/runs/3760086097/jobs/6390434605)) go to building `-html`, `-doc`, etc., as well as `git repack` (see note below). This it not necessarily specific to `aarch64`, as the other architectures also take quite some time.
  - The [`git repack` command](https://github.com/git-for-windows/build-extra/blob/a8464e1aa4a4030175134560b1c985602bc3c07d/please.sh#L4883) takes multiple minutes to complete, but this is the case on all architectures. On the `aarch64` variant, CPU usage was quite constant at 15%, meaning that it was only using 1 out of 8 available cores. Just wanted to bring this to your attention.

The resulting binary works as expected:

```
$ git --version --build-options
git version 2.39.0.windows.2.15.g53a643f669.20221222180905
cpu: aarch64
built from commit: 53a643f66994570dd725f5067ce5f27e5550065f
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

**Comment by dscho** [comment]
Thank you, @dennisameling! And yes, the speed is impressive.

**Comment by dscho** [comment]
@dennisameling and here is the new version: https://github.com/git-for-windows/setup-git-for-windows-sdk/releases/tag/v1.6.2

File: main.ts (hunk 1 above)
Comment (dennisameling) [reply main.ts:29]: PR to teach `create-sdk-artifact` to use `architecture` instead of `bitness`: https://github.com/git-for-windows/build-extra/pull/462

**Pull request merged**